prov/efa/src/efa_base_ep.h

```diff
@@ -33,36 +33,22 @@
 #ifndef EFA_BASE_EP_H
 #define EFA_BASE_EP_H
 
-#include "config.h"
-
-#include <asm/types.h>
-#include <errno.h>
-#include <fcntl.h>
 #include <arpa/inet.h>
-#include <netinet/in.h>
-#include <poll.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
-#include <unistd.h>
-#include <assert.h>
-#include <pthread.h>
-#include <sys/epoll.h>
-
-#include <rdma/fabric.h>
-#include <rdma/fi_cm.h>
-#include <rdma/fi_domain.h>
-#include <rdma/fi_endpoint.h>
-#include <rdma/fi_errno.h>
-
 #include <infiniband/verbs.h>
 #include <infiniband/efadv.h>
 
 #include "ofi.h"
-#include "ofi_enosys.h"
-#include "ofi_list.h"
 #include "ofi_util.h"
-#include "ofi_file.h"
+
+#include "efa_av.h"
```

Comment:
Is this still needed here?

Reply:
yes. `efa_base_ep.c` isn't importing `efa_av.h`

also, if we're going to be so careful with imports, we need to spend a week cleaning up the rest of the code base

Reply:
> yes. `efa_base_ep.c` isn't importing `efa_av.h`

Does `efa.h` not include `efa_av.h`?

> also, if we're going to be so careful with imports

It's not really about carefulness in the sense that it's a correctness thing (it will function just fine); it's more of a code quality thing.

Reply:
> Does `efa.h` not include `efa_av.h`?

Not directly, no. Is this your end goal: `efa.h` includes every DGRAM header and all DGRAM C files only include `efa.h`? That's currently not the case. We can make a task to clean up the includes in this style, if you wish.